pkg/node/manager/manager.go

```diff
@@ -474,11 +474,15 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 	resource := ipcacheTypes.NewResourceID(ipcacheTypes.ResourceKindNode, "", n.Name)
```

Comment:
>                     // 1) We add the IP to the ipset
>                    if m.conf.NodeIpsetNeeded() && address.Type == addressing.NodeInternalIP {
>                            m.ipsetMgr.AddToNodeIpset(address.IP)
>                    }
>
>                    // 2) Say we skip the ipcache update here based on config...
>                    if m.nodeAddressSkipsIPCache(address) {
>                            continue
>                    }
>
>                    ...
>
>                    // 3) ... Then we don't add the IP to nodeIPsAdded, even though we added it to the ipset
>                    nodeIPsAdded = append(nodeIPsAdded, prefix)

Assuming I understand correctly, the natural fix would be to move (1) and (3) together, either before or after (2). Why doesn't that work?

Reply:
Moving 3) towards 1) (before 2) would not work, because we could have `nodeIPsAdded` updated even if we don't update the ipcache entries for these IPs. This is why I use a separate IP list in my current commit.

Moving 1) towards 3) would work to fix the issue, but it would change the current behaviour, and I was not sure if this was the correct thing to do. In particular, if we do skip the ipcache update, the current behaviour adds an ipset entry anyway (the current fix preserves that). If we move 1) towards 3), skipping the ipcache update will mean skipping the ipset entry update as well. I'm struggling to understand whether this is desirable or not.

Reply:
Thanks for the context! I think that would be valuable information to have in the commit message.

> In particular, if we do skip the ipcache update, the current behaviour adds an ipset entry anyway (the current fix preserves that).

I think that's the correct behavior. We want to skip masquerading for NodeInternalIP (i.e., add it to ipset) regardless of whether e.g. enable-remote-node-identity is enabled or not.

Reply:
OK, sounds right indeed.

And you're right that it's worth adding the the description, I updated my commit log (see the first list under the trimmed code).